lib/rubocop/cop/style/hash_syntax.rb

```diff
@@ -96,7 +96,13 @@ def word_symbol_pair?(pair)
           return false unless key.sym_type?
 
           sym_name = key.loc.expression.source
-          sym_name !~ /\A:["']|=\z/
+          valid_19_syntax_symbol?(sym_name)
+        end
+
+        def valid_19_syntax_symbol?(sym_name)
+          return false if sym_name =~ /\A:["']/
```

Comment:
@chastell is right - this check should work on every Ruby. If we use specs that are version specific we should simply tag them accordingly.